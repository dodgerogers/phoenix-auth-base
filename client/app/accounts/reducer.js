import { fromJS } from 'immutable';
import { actionTypes } from './constants';

export const initialState = fromJS({
  currentUser: null,
  currentProfile: null,
  userProfiles: [],
});

export default function accountsReducer(state = initialState, action) {
  const { type } = action;

  switch (type) {
    case actionTypes.GET_CURRENT_USER_SUCCESS:
      return state.set('currentUser', fromJS(action.user));
    case actionTypes.GET_CURRENT_USER_PROFILES_SUCCESS:
      return state.set('userProfiles', fromJS(action.profiles));
    case actionTypes.SET_CURRENT_USER_PROFILE:
      const profile = state.userProfiles.find(p => p.id === action.profileId);
      return state.set('currentProfile', profile);
    case actionTypes.GET_CURRENT_USER_FAILURE:
      return state.set('currentUser', initialState.get('currentUser'));
    case actionTypes.GET_CURRENT_USER_PROFILES_FAILURE:
      return state.set('userProfiles', initialState.get('userProfiles'));
    default:
      return state;
  };
}
