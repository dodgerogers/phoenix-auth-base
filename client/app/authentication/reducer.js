import { fromJS } from 'immutable';
import { actionTypes } from './constants';

export const initialState = fromJS({
  currentUser: null,
  accessToken: null,
  refreshing: false,
});

export default function authenticationReducer(state = initialState, action) {
  const { type } = action;

  switch (type) {
    case actionTypes.VERIFY_TOKEN:
    case actionTypes.SET_TOKEN:
      return state.set('accessToken', fromJS(action.accessToken));
    case actionTypes.VERIFY_TOKEN_SUCCESS:
      return state.set('currentUser', fromJS(action.user));
    case actionTypes.AUTHENTICATE_FAILURE:
    case actionTypes.VERIFY_TOKEN_FAILURE:
    case actionTypes.SIGN_OUT_SUCCESS:
      return state.set('accessToken', null).set('currentUser', null);
    case actionTypes.REFRESH_TOKEN_REQUEST:
      return state.set('refreshing', true);
    case actionTypes.REFRESH_TOKEN_SUCCESS:
    case actionTypes.REFRESH_TOKEN_FAILURE:
      return state.set('refreshing', false);    
    default:
      return state;
  };
}
