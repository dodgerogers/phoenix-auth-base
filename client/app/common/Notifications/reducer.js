import { fromJS } from 'immutable';
import { actionTypes, areaIDs } from './constants';
import uid from '../../lib/utils/UidGenerator';


const initialNotification = fromJS({
  level: null,
  message: null,
});

const initialList = fromJS([]);

export const initialState = fromJS({
  [areaIDs.APPLICATION]: initialList,
  [areaIDs.AUTHENTICATION]: initialList,
});

export default function notificationReducer(state = initialState, action) {
  const { type } = action;

  switch (type) {
    case actionTypes.NOTIFY:
      return state.update(getAreaID(action), array => array.push(incrementID(action.notification, array)));
    case actionTypes.DESTROY:
      return state.update(getAreaID(action), list => filterById(list, action.notification));
    default:
      return state;
  };
}

const getAreaID = (action) => {
  return action.id || areaIDs.APPLICATION;
}

const incrementID = (notification, array) => {
  const lastNotification = array.get(-1);
  const id = lastNotification ? lastNotification.get('id') + 1 : 0;
  notification.id = id;

  return fromJS(notification);
}

const filterById = (list, notification) => {
  return list.filter(n => n.get('id') !== notification.get('id'))
}
