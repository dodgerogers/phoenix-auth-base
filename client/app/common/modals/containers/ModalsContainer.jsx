import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import LoginModal from '../../../authentication/components/LoginModal';
import RegistrationFormModal from '../../../authentication/components/RegistrationFormModal';
import ConfirmationModal from '../../../authentication/components/ConfirmationModal';
import { hideModal } from '../actions';
import { modalIds } from '../constants';


const ModalsContainer = (props) => {
  const registeredModals = [
    { component: LoginModal, id: modalIds.LOGIN_MODAL },
    { component: RegistrationFormModal, id: modalIds.REGISTRATION_MODAL },
    { component: ConfirmationModal, id: modalIds.CONFIRMATION_MODAL },
  ];

  function renderModals() {
    return registeredModals.map(modal => {
      const Component = modal.component;
      return (
        <Component
          key={modal.id}
          show={props.modals.get(modal.id)}
          close={() => props.hideModal(modal.id)}
        />
      );
    });
  }

  return (<div className="modals">{renderModals()}</div>);
}

function mapStateToProps(state) {
  return {
    modals: state.modals,
  };
}

ModalsContainer.propTypes = {
  modals: PropTypes.object.isRequired,
};

export { ModalsContainer as PureComponent };
export default connect(mapStateToProps, { hideModal })(ModalsContainer);
