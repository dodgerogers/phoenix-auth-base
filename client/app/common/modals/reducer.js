import { fromJS, Map } from 'immutable';
import { actionTypes } from './constants';

export const initialState = fromJS({});

export default function modalsReducer(state = initialState, action) {
  const { type } = action;

  switch (type) {
    case actionTypes.SHOW_MODAL:
      return showModal(state, action.data.id);
    case actionTypes.HIDE_MODAL:
      return hideModal(state, action.data.id);
    default:
      return state;
  }
}

function showModal(state, id) {
  return state.merge(hideAllModals(state)).set(id, true);
}

function hideModal(state, id) {
  return state.merge(hideAllModals(state)).set(id, false);
}

function hideAllModals(state) {
  return fromJS(state.keySeq().reduce((memo, key) => {
    memo[key] = false;
    return memo;
  }, {}));
}
