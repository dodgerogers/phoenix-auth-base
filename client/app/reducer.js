import { combineReducers } from 'redux';
import { authStateReducer } from 'redux-oauth';
import { reducer as formsReducer } from 'redux-form/immutable'
import modalsReducer from './common/modals/modalsReducer';

const rootReducer = combineReducers({
  auth: authStateReducer,
  form: formsReducer,
  modals: modalsReducer,
});

export default rootReducer;
