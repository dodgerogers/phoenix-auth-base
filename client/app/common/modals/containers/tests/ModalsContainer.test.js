import React from 'react';
import { fromJS } from 'immutable';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import ModalsContainer, { PureComponent } from '../ModalsContainer';
import { initialState } from '../../reducer';


describe('ModalsContainer', () => {
  const fullProps = () => ({
    modals: initialState,
  });

  describe('PureComponent', () => {
    const connectedComponent = (props) => shallow(
      <PureComponent {...props} />
    );

    it('matches snapshot', () => {
      const props = fullProps();
      const component = connectedComponent(props);

      expect(toJson(component)).toMatchSnapshot();
    });
  });
});
