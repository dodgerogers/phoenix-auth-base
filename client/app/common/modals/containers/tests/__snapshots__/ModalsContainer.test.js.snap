// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalsContainer PureComponent matches snapshot 1`] = `
<div
  className="modals"
>
  <LoginModal
    close={[Function]}
  />
  <RegistrationFormModal
    close={[Function]}
  />
  <ConfirmationModal
    close={[Function]}
  />
  <ResendConfirmationModal
    close={[Function]}
  />
  <ForgotPasswordModal
    close={[Function]}
  />
  <ResetPasswordModal
    close={[Function]}
  />
</div>
`;
